app/uk/gov/hmrc/apiplatformoutboundsoap/repositories/OutboundMessageRepository.scala

```diff
@@ -16,32 +16,73 @@
 
 package uk.gov.hmrc.apiplatformoutboundsoap.repositories
 
+import akka.stream.Materializer
+import akka.stream.scaladsl.Source
+import org.joda.time.DateTime
+import org.joda.time.DateTime.now
+import org.joda.time.DateTimeZone.UTC
+import play.api.libs.json.Json
 import play.modules.reactivemongo.ReactiveMongoComponent
+import reactivemongo.akkastream.{cursorProducer}
+import reactivemongo.api.ReadPreference
 import reactivemongo.api.indexes.Index
 import reactivemongo.api.indexes.IndexType.Ascending
-import reactivemongo.bson.BSONObjectID
-import uk.gov.hmrc.apiplatformoutboundsoap.models.OutboundSoapMessage
+import reactivemongo.bson.{BSONDocument, BSONLong, BSONObjectID}
+import reactivemongo.play.json.ImplicitBSONHandlers.JsObjectDocumentWriter
+import uk.gov.hmrc.apiplatformoutboundsoap.models.{OutboundSoapMessage, RetryingOutboundSoapMessage, SendingStatus}
 import uk.gov.hmrc.apiplatformoutboundsoap.repositories.MongoFormatter.outboundSoapMessageFormatter
 import uk.gov.hmrc.mongo.ReactiveRepository
 import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
+import uk.gov.hmrc.apiplatformoutboundsoap.repositories.MongoFormatter.dateFormat
 
+import java.util.UUID
 import javax.inject.{Inject, Singleton}
+import scala.concurrent.duration.Duration
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
 class OutboundMessageRepository @Inject()(mongoComponent: ReactiveMongoComponent)
-                                         (implicit ec: ExecutionContext)
+                                         (implicit ec: ExecutionContext, m: Materializer)
   extends ReactiveRepository[OutboundSoapMessage, BSONObjectID](
     "messages",
     mongoComponent.mongoConnector.db,
     outboundSoapMessageFormatter,
     ReactiveMongoFormats.objectIdFormats) {
 
+  val MESSAGE_TTL: Long = Duration("30 days").toSeconds
```

Comment:
The only problem with this not being configurable is that in QA we'd need to wait for a month to see if it works. But making it configurable is not just a matter of taking the value from the configuration. To change the index, it needs to be dropped and created again.

Reply:
Made configurable but if indexes need to be dropped, this will have to be done manually